app/controllers/users/totp_setup_controller.rb

```diff
@@ -47,5 +46,9 @@ def process_invalid_code
       flash[:error] = t('errors.invalid_totp')
       redirect_to authenticator_setup_path
     end
+
+    def new_totp_secret
+      user_session.fetch(:new_totp_secret, nil)
```

Comment:
fetch with nil as second arg isn't needed right?  just user_session[:new_totp_secret] is clearer in this case, no?